packages/world/src/interfaces/IWorld.sol

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { IStore } from "@latticexyz/store/src/IStore.sol";
+
+import { IWorldCore } from "./IWorldCore.sol";
+
+// TODO: autogenerate / auto-include all system interfaces
+import { IRegistrationSystem } from "./systems/IRegistrationSystem.sol";
+
+interface IWorld is IStore, IWorldCore, IRegistrationSystem {}
```

Comment:
@dk1a what do you think about adding an autogen function to create interfaces for systems that are registered in the World (defined via `mud.config.mts`, then registered via `registerFunctionSelector` / `registerRootFunctionSelector`)?

Reply:
Is there a limit to how many inheritance a contract can have?

Reply:
I don't _think_ there is, but haven't found explicit documentation about that

Reply:
> @dk1a what do you think about adding an autogen function to create interfaces for systems that are registered in the World (defined via `mud.config.mts`, then registered via `registerFunctionSelector` / `registerRootFunctionSelector`)?

Something like `worldgen`. I think it's necessary for being able to interact with systems on-chain, and maybe helps the client too. Doing it manually would be a huge hassle with lots of room for error.

I don't think systems should actually implement their own interface, it'd just be generated along with `World` based on manually defined public/external funcs